src/SpeciesElements.f90

```diff
@@ -0,0 +1,167 @@
+!>
+   FUNCTION getElementArray(this, formula) RESULT(eleArray)
+      CLASS(SpeciesElementsType),INTENT(INOUT) :: this
+      type(StringType), INTENT(IN) :: formula
+
+      REAL(SDK), dimension(0:118) :: eleArray ! Return Value
+      INTEGER(SIK) :: i, Z
+      REAL(SDK) :: coeff
+      CHARACTER(LEN=1) :: letter, nextLetter
+      CHARACTER(LEN=2) :: element
+      TYPE(ElementsIsotopesType) :: myEI
+
+      eleArray = 0.0
+
+      DO i=1, LEN(formula)
+         coeff = 1.0
+         ! Gets the value in the formula
+         letter = CHAR(formula,i,i)
+         ! Test to see if its a letter         
+         IF (this%isChar(letter)) THEN
+            ! Test to see if its capital
+            IF (this%isCharCap(letter)) THEN
+               element = letter
+               IF (i /= LEN(formula)) THEN
+                  ! Gets the next letter to see if its lower or upper.
+                  nextLetter = CHAR(formula,i+1,i+1)
+               ELSE
+                  nextLetter = 'A'
+               END IF
+               ! If its lower case then it needs to be added to the 
+               ! current letter
+               IF (this%isCharLow(nextLetter)) THEN
+                  element = letter // nextLetter
+               END IF
+            END IF
+         ! If the character is lower case then do nothing
+         ELSE IF (this%isCharLow(letter)) THEN
+            CONTINUE 
+         ! If not a letter then its a stoichiometrix coefficient
+         ELSE
+            coeff = letter
+         END IF
+         ! Gets the atomic number
+         Z = myEI%getAtomicNumber(element)
+         ! Sets the coeff in the array
+         eleArray(Z) = coeff 
+      END DO
+   ENDFUNCTION  getElementArray
+!
+!-------------------------------------------------------------------------------
+!> @brief Determins whether a character is in the alphabet using ASCII format
+!> @param letter character to check
+!>
+   FUNCTION isChar(this, letter) RESULT(isValid)
+      CLASS(SpeciesElementsType),INTENT(INOUT) :: this
+      CHARACTER(LEN=*), INTENT(IN) :: letter
+
+      LOGICAL(SBK) :: isValid
+      INTEGER(SIK) :: val
+
+      val = IACHAR(letter)
+
+      isValid = .FALSE.
+      ! Checks against the upper case letters
+      IF (65<= val .AND. val<=90) THEN
+         isValid = .TRUE.
+      ! Checks against the lower case letters
+      ELSE IF  (97<= val .AND. val<=122) THEN
+         isValid = .TRUE.
+      end if 
+
+   ENDFUNCTION isChar 
+!
+!-------------------------------------------------------------------------------
+!> @brief Determins whether a character is capitilized using ASCII format
+!> @param letter character to check
+!>
+   FUNCTION isCharCap(this, letter) RESULT(isValid)
+      CLASS(SpeciesElementsType),INTENT(INOUT) :: this
+      CHARACTER(LEN=*), INTENT(IN) :: letter
+
+      LOGICAL(SBK) :: isValid
+      INTEGER(SIK) :: val
+
+      val = IACHAR(letter)
+
+      isValid = .FALSE.
+      IF (65<= val .AND. val<=90) THEN
```

Comment:
We generally leave a space on either side of binary logical operators... so `65 <= val .AND. val <= 90`.  Check that in the following routine too.